### PR TITLE
Make mgmt release a job dependent on ShouldPublish parameter

### PR DESCRIPTION
**Pull request by chidozieononiwu**
- Make mgmt release a job
- Remove the need to approve release and just make it depend on the ShouldPublish parameter

### CODE REVIEW

File: eng/pipelines/templates/jobs/mgmt-release.yml [hunk 1]
```diff
@@ -12,76 +9,70 @@ parameters:
...
-                        /p:MSPublicSymbolsPAT=$(azuresdk-microsoftpublicsymbols-devops-pat)
-                        /p:MSSymbolsPAT=$(azuresdk-microsoft-devops-pat)
-                        /p:AzureSDKSymbolsPAT=$(azuresdk-azure-sdk-devops-pat)
-                    condition: and(succeeded(), eq(variables['Build.Repository.Name'], 'Azure/azure-sdk-for-net'))
+      - task: MSBuild@1
+        displayName: 'Upload Symbols'
+        inputs:
+          solution: '${{ parameters.BuildToolsPath }}/tools/symboltool/SymbolUploader.proj'
+          msbuildArguments: >
+            /p:PackagesPath=$(PIPELINE.WORKSPACE)/packages/Azure.Template
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/mgmt-release.yml:54]: Why did you add "Azure.Template" here? Was that leftover from testing?

File: eng/pipelines/mgmt.yml [hunk 2]
```diff
@@ -202,9 +202,7 @@ stages:
...
           ShouldPublish: ${{ parameters.ShouldPublish }}
 
-  # The Release job is conditioned on whether we are building a pull request and the branch.
-  - ${{ if and(ne(variables['Build.Reason'], 'PullRequest'), eq(variables['System.TeamProject'], 'internal')) }}:
-    - template: /eng/pipelines/templates/jobs/mgmt-release.yml
-      parameters:
-        ShouldPublish: ${{ parameters.ShouldPublish }}
-        DependsOn: Build
\ No newline at end of file
+      - ${{ if and(ne(variables['Build.Reason'], 'PullRequest'), eq(parameters.ShouldPublish, 'true')) }}:
```
Comment (weshaggard) [inline eng/pipelines/mgmt.yml:205]: I don't think this will be consume in the public but just in case I think it is worth keeping the internal check.

File: eng/pipelines/templates/jobs/mgmt-release.yml (hunk 1 above)
Comment (chidozieononiwu) [reply eng/pipelines/templates/jobs/mgmt-release.yml:54]: Yes, I used the data plane template package to test it.

**Review by weshaggard — APPROVED** [review]

File: eng/pipelines/mgmt.yml [hunk 3]
```diff
@@ -202,9 +202,7 @@ stages:
...
           ShouldPublish: ${{ parameters.ShouldPublish }}
 
-  # The Release job is conditioned on whether we are building a pull request and the branch.
-  - ${{ if and(ne(variables['Build.Reason'], 'PullRequest'), eq(variables['System.TeamProject'], 'internal')) }}:
-    - template: /eng/pipelines/templates/jobs/mgmt-release.yml
-      parameters:
-        ShouldPublish: ${{ parameters.ShouldPublish }}
-        DependsOn: Build
\ No newline at end of file
+      - ${{ if and(ne(variables['Build.Reason'], 'PullRequest'), eq(parameters.ShouldPublish, 'true'),  eq(variables['System.TeamProject'], 'internal')) }}:
```
Comment (weshaggard) [inline eng/pipelines/mgmt.yml:205]:
```suggestion
      - ${{ if and(eq(parameters.ShouldPublish, 'true'), ne(variables['Build.Reason'], 'PullRequest'), eq(variables['System.TeamProject'], 'internal')) }}:
```
